### PR TITLE
fix(github): modify bp v200

### PR DESCRIPTION
**Pull request by warren830**
### Summary
Change the way to set options in plugin GitHub.
Before, we set all values to tasks.GithubOptions when making plan, right now, we just pass scopeId to options, and set other values when preparing plugin data

### Does this close any open issues?
relate to #3468 

### Screenshots
we can see that the blueprint can be triggered successfully
image

bp V100:
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/api/blueprint_v200.go [hunk 1]
```diff
@@ -123,39 +125,37 @@ func makeDataSourcePipelinePlanV200(
...
 			}
-			stage, err = addGitex(bpScope.Entities, connection, repo, stage)
+			stage, err = addGitex(bpScope.Entities, connection, repoRes, stage)
 			if err != nil {
 				return nil, err
 			}
 		}
 		plan[i] = stage
 	}
+	fmt.Println(plan)
```
Comment (likyh) [inline plugins/github/api/blueprint_v200.go:158]: delete debug code

File: plugins/github/impl/impl.go [hunk 2]
```diff
@@ -159,6 +161,32 @@ func (plugin Github) PrepareTaskData(taskCtx core.TaskContext, options map[strin
 	if err != nil {
 		return nil, errors.Default.Wrap(err, "unable to get github connection by the given connection ID")
 	}
+	var githubRepo models.GithubRepo
+	// for advanced mode which we already set value to onwer/repo
+	if op.Owner != "" && op.Repo != "" {
+		err = taskCtx.GetDal().First(&githubRepo, dal.Where(
+			"connection_id = ? AND name = ? AND owner_login = ",
```
Comment (likyh) [inline plugins/github/impl/impl.go:168]: it should be "owner_login = ?"

File: plugins/github_graphql/impl/impl.go [hunk 3]
```diff
@@ -137,6 +133,41 @@ func (plugin GithubGraphql) PrepareTaskData(taskCtx core.TaskContext, options ma
 		return nil, errors.Default.Wrap(err, "unable to get github connection by the given connection ID: %v")
 	}
 
+	var githubRepo models.GithubRepo
+	// for advanced mode or others which we already set value to onwer/repo
+	if op.Owner != "" && op.Repo != "" {
+		err = taskCtx.GetDal().First(&githubRepo, dal.Where(
+			"connection_id = ? AND name = ? AND owner_login = ",
```
Comment (likyh) [inline plugins/github_graphql/impl/impl.go:140]: =?

Comment (warren830) [reply plugins/github_graphql/impl/impl.go:140]: fixed

File: plugins/github/impl/impl.go (hunk 2 above)
Comment (warren830) [reply plugins/github/impl/impl.go:168]: fixed

File: plugins/github/api/blueprint_v200.go [hunk 4]
```diff
@@ -86,9 +85,12 @@ func makeDataSourcePipelinePlanV200(
 	apiClient helper.ApiClientGetter,
 ) (core.PipelinePlan, errors.Error) {
 	var err errors.Error
-	var stage core.PipelineStage
-	var repo *tasks.GithubApiRepo
+	var repoRes *tasks.GithubApiRepo
 	for i, bpScope := range bpScopes {
+		stage := plan[i]
```
Comment (likyh) [inline plugins/github/api/blueprint_v200.go:90]: plan and bpScopes have the same order and count?

File: plugins/github_graphql/impl/impl.go [hunk 5]
```diff
@@ -178,6 +178,49 @@ func (plugin GithubGraphql) PrepareTaskData(taskCtx core.TaskContext, options ma
 	}, nil
 }
 
+func EnrichOptions(taskCtx core.TaskContext,
```
Comment (likyh) [inline plugins/github_graphql/impl/impl.go:181]: Defined in github_graphql?

File: plugins/github_graphql/impl/impl.go [hunk 6]
```diff
@@ -178,6 +178,49 @@ func (plugin GithubGraphql) PrepareTaskData(taskCtx core.TaskContext, options ma
...
+		err := taskCtx.GetDal().First(&githubRepo, dal.Where(
+			"connection_id = ? AND name = ? AND owner_login = ?",
+			op.ConnectionId, op.Repo, op.Owner))
+		if err != nil {
+			return nil, errors.Default.Wrap(err, fmt.Sprintf("fail to find repo %s/%s", op.Owner, op.Repo))
+		}
+		op.TransformationRuleId = githubRepo.TransformationRuleId
+	}
+	// for bp v200 which we only set ScopeId for options
+	if &githubRepo == nil && op.ScopeId != "" {
```
Comment (likyh) [inline plugins/github_graphql/impl/impl.go:198]: It looks like `&githubRepo == nil` always true.

Comment (warren830) [reply plugins/github_graphql/impl/impl.go]: > It looks like `&githubRepo == nil` always true.

yes, already fixed

**Pull request merged**